yarn-project/cli/src/cmds/infrastructure/sequencers.ts

```diff
@@ -71,24 +74,62 @@ export async function sequencers(opts: {
       client: walletClient,
     });
 
-    await Promise.all(
-      [
-        await stakingAsset.write.mint([walletClient.account.address, MINIMUM_STAKE], {} as any),
-        await stakingAsset.write.approve([rollup.address, MINIMUM_STAKE], {} as any),
-      ].map(txHash => publicClient.waitForTransactionReceipt({ hash: txHash })),
-    );
+    const mintRequest = {
+      to: stakingAsset.address,
+      data: encodeFunctionData({
+        abi: stakingAsset.abi,
+        functionName: 'mint',
+        args: [walletClient.account.address, MINIMUM_STAKE],
+      }),
+    };
+    const approveRequest = {
+      to: stakingAsset.address,
+      data: encodeFunctionData({
+        abi: stakingAsset.abi,
+        functionName: 'approve',
+        args: [rollup.address, MINIMUM_STAKE],
+      }),
+    };
+
+    // Send transactions first
+    const [mintTxRes, approveTxRes] = await Promise.all([
+      l1TxUtils.sendTransaction(mintRequest),
+      l1TxUtils.sendTransaction(approveRequest),
+    ]);
+
+    // Monitor transactions in parallel
+    await Promise.all([
+      l1TxUtils.monitorTransaction(mintRequest, mintTxRes.txHash, { gasLimit: mintTxRes.gasLimit }),
+      l1TxUtils.monitorTransaction(approveRequest, approveTxRes.txHash, { gasLimit: approveTxRes.gasLimit }),
+    ]);
```

Comment:
Curious: why not a single `Promise.all` with `sendAndMonitor`?

Reply:
I feel like I had a reason and now realizing that it was probably invalid since we're sending these 2 at the same time anyway